Automattic/jetpack
# Show HLS Playlist for the VideoPress video url, instead of fragmented mp4 file

This PR exposes the HLS playlist as the video url for all videos that were uploaded and transcoded using the new transcode process (default for all videopress uploads, no options for not transcoding this way). This change is in response to some support requests where 3rd party video players were being used on a site, instead of the VideoPress video player (which isn't a scenario that we directly support, but seems to have been working fine until we started working on the video streaming improvements).

A user's process for configuring the 3rd party player was to copy the video url from the Media Library and provide it to the player. The result of this was the video file was not playing until the entire file was downloaded since it is a fragmented mp4 file (due to the new transcode process). The implication of this change is that the 3rd party player will now need to support HLS playlists in order to play the video.

<img width="433" alt="Screen Shot 2021-08-26 at 1 42 46 PM" src="https://user-images.githubusercontent.com/4081020/131010524-f8253a82-fe58-43f1-a328-23af46279d8d.png">

Fixes 931-gh-Automattic/greenhouse

#### Changes proposed in this Pull Request:

* If an HLS file is present in the VideoPress video's metadata, expose it as the "Url" in the media library instead of the mp4 file. The mp4 file is a fragmented file that won't play in the browser until it is fully downloaded.
* If an HLS file is not present (for all older VideoPress videos) then the mp4 file will be exposed as the "Url" in the media library.

#### Jetpack product discussion


Not a Jetpack Product discussion, but a description of the issue and discussion with HEs 928-gh-Automattic/greenhouse
If this seems like something that needs a larger discussion, I'm happy to start a P2 post for it.

#### Does this pull request change what data or activity we track or use?
nope

#### Testing instructions:






* On a Jetpack site with VideoPress enabled, visit the Media Library
* Upload a new video (or open a recently uploaded video). The url property should be a url that ends in `.master.m3u8` if it was uploaded recently (the last month or so depending on the type of user being tested with)
* For older videos, a url ending in `.mp4` should be displayed
